tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/tests/InstanceMethods.java

```diff
@@ -172,4 +176,10 @@ public int getInt(ObjectValue objectValue, int x) {
     public int getRandomInt() {
         return 123;
     }
+
+    public BString getCurrentModule(Environment env, long b) {
+        Module callerModule = env.getCurrentModule();
+        return StringUtils.fromString(callerModule.getOrg() + "#" + callerModule.getName() + "#" +
+                                              callerModule.getVersion() + "#" + b);
+    }
```

Comment:
Shall we also have test where self is also passed as a parameter for an instance method? I mean self and Environment both.

Reply:
Shall we also see what happens if the methods with both without self and with self and Environment have the same name(overloaded)?

Reply:
Self is not part of parameter list in instance method. As an example getCureentModule(receiver) will generated to receiver.getCurrentModule() . passing self as a parameter test already added. I also added another test case for overload methods.